app/screens/channel_add_people/channel_add_people.tsx

```diff
@@ -0,0 +1,340 @@
+    currentChannel: ChannelModel;
+    currentTeamId: string;
+    currentUserId: string;
+    restrictDirectMessage: boolean;
+    teammateNameDisplay: string;
+    tutorialWatched: boolean;
+}
+
+export default function ChannelAddPeople({
+    componentId,
+    currentChannel,
+    currentTeamId,
+    currentUserId,
+    restrictDirectMessage,
+    teammateNameDisplay,
+    tutorialWatched,
+}: Props) {
+    const serverUrl = useServerUrl();
+    const intl = useIntl();
+    const theme = useTheme();
+    const {formatMessage} = intl;
+
+    const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
+    const next = useRef(true);
+    const page = useRef(-1);
+    const mounted = useRef(false);
+
+    const [profiles, dispatchProfiles] = useReducer(reduceProfiles, []);
+    const [searchResults, setSearchResults] = useState<UserProfile[]>([]);
+    const [loading, setLoading] = useState(false);
+    const [term, setTerm] = useState('');
+    const [startingConversation, setStartingConversation] = useState(false);
+    const [selectedIds, setSelectedIds] = useState<{[id: string]: UserProfile}>({});
+    const selectedCount = Object.keys(selectedIds).length;
+    const groupConstrained = currentChannel.isGroupConstrained;
+    const currentChannelId = currentChannel.id;
+
+    const isSearch = Boolean(term);
+
+    const loadedProfiles = ({users}: {users?: UserProfile[]}) => {
+        if (mounted.current) {
+            if (users && !users.length) {
+                next.current = false;
+            }
+
+            page.current += 1;
+            setLoading(false);
+            dispatchProfiles({type: 'add', values: users});
+        }
+    };
+
+    const getProfiles = useCallback(debounce(() => {
+        if (next.current && !loading && !term && mounted.current) {
+            setLoading(true);
+            fetchProfilesNotInChannel(serverUrl,
+                currentTeamId,
+                currentChannelId,
+                groupConstrained,
+                page.current + 1,
+                General.PROFILE_CHUNK_SIZE).then(loadedProfiles);
+        }
+    }, 100), [loading, isSearch, serverUrl, currentTeamId]);
+
+    const handleRemoveProfile = useCallback((id: string) => {
+        const newSelectedIds = Object.assign({}, selectedIds);
+
+        Reflect.deleteProperty(newSelectedIds, id);
+
+        setSelectedIds(newSelectedIds);
+    }, [selectedIds]);
+
+    const addPeopleToChannel = useCallback(async (ids: string[]): Promise<boolean> => {
+        const result = await addMembersToChannel(serverUrl, currentChannelId, ids, '', false);
+
+        if (result.error) {
+            alertErrorWithFallback(intl, result.error, messages.error);
+        }
+
+        return !result.error;
+    }, [serverUrl]);
+
+    const clearSearch = useCallback(() => {
+        setTerm('');
+        setSearchResults([]);
+    }, []);
+
+    const startConversation = useCallback(async (selectedId?: {[id: string]: boolean}) => {
```

Comment:
This function is called startConversation (which I would expect it to send me to a dm) but what it actually does is adding a person to the channel. Should this function be renamed, or I am missing something?